### PR TITLE
Issue 6

### LINKED ISSUE #6: As a developer, I want a Doxygen config file so that I can generate documentation

### PR DESCRIPTION
**Pull request by fosterj14**
Created the doxygen config file. This will generate documentation and place it in the src folder when ran from the src directory. We may want to integrate this into CMake at a later point in order to automate the documentation process when building the project.

### CODE REVIEW

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2494 @@
...
+
+DOXYFILE_ENCODING      = UTF-8
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "Craft"
```
Comment (lukemoore495) [inline Doxyfile]: Maybe we ought to call the Project "Remainders-Craft" to match the repository.

File: Doxyfile [hunk 2]
```diff
@@ -0,0 +1,2494 @@
...
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         =
+
+# Using the PROJECT_BRIEF tag one can provide an optional one line description
+# for a project that appears at the top of each page and should give viewer a
+# quick idea about the purpose of the project. Keep the description short.
+
+PROJECT_BRIEF          =
```
Comment (lukemoore495) [inline Doxyfile]: It might be good to link the GitHub repository's ReadMe here or some brief description of the project.

**Review by lukemoore495 — CHANGES_REQUESTED** [review]

File: Doxyfile (hunk 1 above)
Comment (fosterj14) [reply Doxyfile]: Changed the name

File: Doxyfile (hunk 2 above)
Comment (fosterj14) [reply Doxyfile]: Added info from the ReadMe

**Review by TiMcCall — APPROVED** [review]
These changes have been reviewed and look to be in good order. The documentation was concise and explained things nicely.

**Review by lukemoore495 — APPROVED** [review]

**Pull request merged**